pkg/kubelet/kubelet.go

```diff
@@ -273,11 +275,30 @@ func makePodSourceConfig(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps
 		glog.Infof("Adding manifest url %q with HTTP header %v", kubeCfg.ManifestURL, manifestURLHeader)
 		config.NewSourceURL(kubeCfg.ManifestURL, manifestURLHeader, nodeName, kubeCfg.HTTPCheckFrequency.Duration, cfg.Channel(kubetypes.HTTPSource))
 	}
+
+	// This is a one-shot fill the pipeline on startup if checkpoints exist
+	if kubeCfg.CheckpointPath != "" {
+		glog.Infof("Adding checkpoint path: %v", kubeCfg.CheckpointPath)
+		cmgr = checkpoint.NewCheckpointManager(kubeCfg.CheckpointPath)
+		c := cfg.Channel(kubetypes.ApiserverSource)
+		pods, err := cmgr.LoadPods()
+		if err != nil {
+			return nil, nil, err
+		}
+		if len(pods) > 0 {
+			c <- kubetypes.PodUpdate{Pods: pods, Op: kubetypes.SET, Source: kubetypes.ApiserverSource}
```

Comment:
won't this make the kubelet kill any currently running containers not in the checkpointed list?

Reply:
@liggitt is right. Kubelet assumes the first update from a source includes the complete list of pods. Once all sources are ready, GC will kick in to delete containers that do not belong to a known pod.

Wiring this so that kubelet considers the source is *not ready* is required.